dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/collections/EvictingQueue.kt

```diff
@@ -0,0 +1,104 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+package com.datadog.android.core.collections
+
+import java.util.LinkedList
+
+/**
+ * A bounded queue that automatically evicts the oldest elements when new elements are added beyond its maximum capacity.
+ *
+ * This class extends [LinkedList] to provide a FIFO (first-in, first-out) queue with a fixed maximum size.
+ * When adding new elements, if the queue is already at capacity, the oldest element is automatically removed.
+ *
+ * @param T the type of elements held in this collection.
+ * @param maxSize the maximum number of elements the queue can hold. Must be greater than 0.
+ *                The default value is [Int.MAX_VALUE], which effectively means there is no practical bound.
+ *
+ * @throws IllegalArgumentException if [maxSize] is less than or equal to 0.
+ */
+class EvictingQueue<T>(private val maxSize: Int = Int.MAX_VALUE) : LinkedList<T>() {
+
+    init {
+        @Suppress("UnsafeThirdPartyFunctionCall") // this line should throw an exception
+        require(maxSize > 0) { "maxSize should be > 0" }
```

Comment:
maybe instead of throwing we can log a message and create a queue with 0 size?

Reply:
I changed implementation of EvictingQueue a bit, so it's does not throwing any exception now